test/unittest/base/test_datalayer.py

```diff
@@ -9,16 +9,136 @@
     torch = None
 
 
+import dataclasses as dc
+from unittest.mock import patch
+
 from superduperdb.backends.mongodb.query import Collection
+from superduperdb.base.artifact import Artifact
 from superduperdb.base.document import Document
 from superduperdb.base.exceptions import ComponentInUseError, ComponentInUseWarning
+from superduperdb.components.component import Component
 from superduperdb.components.dataset import Dataset
 from superduperdb.components.encoder import Encoder
 from superduperdb.components.listener import Listener
 
 n_data_points = 250
 
 
+@dc.dataclass
+class TestComponent(Component):
+    identifier: str
+    version: int = None
+    type_id = 'test-component'
+    is_on_create = False
+    is_on_load = False
+    is_schedule_jobs = False
+    child: 'TestComponent' = None
+    artifact: Artifact = None
+
+    def on_create(self, db):
+        self.is_on_create = True
+
+    def on_load(self, db):
+        self.is_on_load = True
+
+    def schedule_jobs(self, db, dependencies):
+        self.is_schedule_jobs = True
+        return []
+
```

Comment:
should we create a component with artifact that is not seriablizable and try to add it to db 

this could be a test

Reply:
Good idea

Reply:
1. Create UT for this case 
2. notify a nice error  message to the user about the artifact not be serializable

Reply:
I added a ut `test_add_compenent_with_bad_artifact` and modified the error message